grpc/src/main/java/io/stargate/grpc/service/interceptors/NewConnectionInterceptor.java

```diff
@@ -73,6 +74,11 @@ public NewConnectionInterceptor(
   @Override
   public <ReqT, RespT> Listener<ReqT> interceptCall(
       ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+    // Allow reflection calls through without authentication
+    if (call.getMethodDescriptor().getServiceName() != null
+        && call.getMethodDescriptor().getServiceName().equals(ServerReflectionGrpc.SERVICE_NAME)) {
+      return next.startCall(call, headers);
+    }
```

Comment:
anybody having an idea if this would be in any case a security issue?

Reply:
I cannot think of obvious significant problems.

What I can think of is that allowing reflection does give out more information. It could perhaps also allow DoS approach if reflection calls are much more expensive to handle than failing authentication.
But I don't see either of these as a practical problem. And at least we have easy switch for turning it off if necessary.

Reply:
Good points..